gst_plugins/python/frame_tag_filter.py

```diff
@@ -0,0 +1,233 @@
+
+        self.sink_pad.set_chain_function_full(self.handle_buffer)
+        self.sink_pad.add_probe(
+            Gst.PadProbeType.EVENT_DOWNSTREAM,
+            on_pad_event,
+            {Gst.EventType.CAPS: self.on_caps},
+        )
+
+    def do_state_changed(self, old: Gst.State, new: Gst.State, pending: Gst.State):
+        """Validate properties when the element is started."""
+
+        init_state = [Gst.State.NULL, Gst.State.READY]
+        if old in init_state and new not in init_state:
+            try:
+                required_property(self.source_id, 'source-id')
+                required_property(self.tag, 'tag')
+            except RequiredPropertyError as exc:
+                self.logger.exception('Failed to start element: %s', exc, exc_info=True)
+                frame = inspect.currentframe()
+                propagate_gst_setting_error(self, frame, __file__, text=exc.args[0])
+
+    def do_get_property(self, prop):
+        """Get property callback."""
+
+        if prop.name == 'tag':
+            return self.tag
+        if prop.name == 'source-id':
+            return self.source_id
+        raise AttributeError(f'Unknown property {prop.name}')
+
+    def do_set_property(self, prop, value):
+        """Set property callback."""
+
+        self.logger.debug('Setting property "%s" to "%s".', prop.name, value)
+        if prop.name == 'tag':
+            self.tag = value
+        elif prop.name == 'source-id':
+            self.source_id = value
+        else:
+            raise AttributeError(f'Unknown property {prop.name}')
+
+    def handle_buffer(
+        self,
+        sink_pad: Gst.Pad,
+        element: Gst.Element,
+        buffer: Gst.Buffer,
+    ) -> Gst.FlowReturn:
+        """Route buffer either to src_tagged or src_not_tagged pad."""
+
+        self.logger.debug('Handling buffer PTS=%s', buffer.pts)
+
+        not_tagged_buffers = self.parse_buffer(buffer)
+        tagged = not_tagged_buffers is None
+        if tagged != self.last_tagged:
+            self.logger.debug('Switching to tagged=%s', tagged)
+            event = build_stream_part_event(next(self.part_id_gen), tagged)
+            self.src_pad_not_tagged.push_event(event)
+            self.src_pad_tagged.push_event(event)
+
+        self.last_tagged = tagged
+        if tagged:
+            return self.push_buffer(self.src_pad_tagged, buffer)
+
+        for not_tagged_buffer in not_tagged_buffers:
+            ret = self.push_buffer(self.src_pad_not_tagged, not_tagged_buffer)
+            if ret != Gst.FlowReturn.OK:
+                self.logger.error(
+                    'Failed to push buffer PTS=%s: %s',
+                    not_tagged_buffer.pts,
+                    ret,
+                )
+                return ret
+
+        return Gst.FlowReturn.OK
+
+    def parse_buffer(self, buffer: Gst.Buffer) -> Optional[List[Gst.Buffer]]:
+        """Parse buffer and return list of not tagged buffers or None if buffer is tagged."""
```

Comment:
Please add a comment, that the DS batch is expected to have only 1 frame at this moment, and `non_tagged_buffers` list is just a precaution